crates/ruff/src/resolver.rs

```diff
@@ -327,10 +317,24 @@ pub fn python_files_in_path(
             }
 
             if result.as_ref().map_or(true, |entry| {
-                // Accept all files that are passed-in directly.
-                (entry.depth() == 0 && entry.file_type().map_or(false, |ft| ft.is_file()))
-                    // Accept all Python files.
-                    || is_python_entry(entry)
+                if entry.depth() == 0 {
+                    // Accept all files that are passed-in directly.
+                    entry.file_type().map_or(false, |ft| ft.is_file())
+                } else {
+                    // Otherwise, check if the file is included.
+                    let path = entry.path();
+                    let resolver = resolver.read().unwrap();
+                    let settings = resolver.resolve(path, pyproject_strategy);
+                    if settings.include.is_match(path) {
+                        debug!("Included path via `include`: {:?}", path);
+                        true
+                    } else if settings.extend_include.is_match(path) {
```

Comment:
It's nice to have the separate debug "reason" but I agree, in theory at least it would be faster. In practice (and in my benchmarks), `extend_include` is almost always empty.